app/file_handlers/pdf_handler.py

```
@@ -0,0 +1,74 @@
+    Raises:
+        FileNotFoundError: Если файл не найден
+        ValueError: Если файл не является PDF или не может быть прочитан
+        IOError: При ошибках ввода/вывода
+    """
+    # Если передан путь к файлу, открываем его
+    if isinstance(file, (str, Path)):
+        file_path = Path(file)
+        if not file_path.exists():
+            logger.error(f"File not found: {file_path}")
+            raise FileNotFoundError(f"File not found: {file_path}")
+        file = open(file_path, 'rb')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)